cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/KafkaCruiseControlServletUtils.java

```diff
@@ -354,6 +359,12 @@ static String urlDecode(String s) throws UnsupportedEncodingException {
 
   static List<String> getGoals(HttpServletRequest request) throws UnsupportedEncodingException {
     boolean isKafkaAssignerMode = getMode(request);
+    EndPoint endPoint = endPoint(request);
+    if (endPoint == FIX_OFFLINE_REPLICAS && isKafkaAssignerMode) {
+      throw new IllegalArgumentException(
+          String.format("The endpoint %s cannot be used in Kafka Assigner mode.", endPoint.name()));
+    }
+
```

Comment:
I think it is better to not let `FIX_OFFLINE_REPLICAS` to support `kafka_assigner_mode` parameter in the first place.